new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Domain.java

```diff
@@ -1,26 +1,35 @@
-// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Copyright 2020, Oracle Corporation and/or its affiliates.
 // Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 
 package oracle.weblogic.kubernetes.assertions.impl;
 
-import java.util.Random;
 import java.util.concurrent.Callable;
 
-import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
-
 public class Domain {
 
-    public static Callable<Boolean> exists(String domainUID, String namespace) {
-        return () -> {
-            int outcome = new Random(System.currentTimeMillis()).nextInt(3);
-            if (outcome == 1) {
-                logger.info(String.format("Domain %s exists in namespace %s", domainUID, namespace));
-                return true;
-            } else {
-                logger.info(String.format("Domain %s does not exist in namespace %s", domainUID, namespace));
-                return false;
-            }
-        };
-    }
+  public static Callable<Boolean> exists(String domainUID, String namespace) {
+    return () -> {
+      String pods[] = {};
+      for (String pod : pods) {
+        if (!Kubernetes.podExists(pod, domainUID, namespace).call().booleanValue()) {
+          return false;
+        }
+      }
+      for (String pod : pods) {
```

Comment:
is this a copy/paste error?

Reply:
what error ?

Reply:
method is called exists() 
you first loop through and see if they exist, then you loop through again and check if they are running.
was wondering why that second loop is there? seemed like maybe it was copy/pasted from another method?

Reply:
The existing tests check if the pod is getting created but not ready yet and then checks if it is running. If the assertion is only to check if the domain and pods exist (in ready state) then we may not need the podExists check. Removed it.

Reply:
ok great, was just checking, since it did not seem to match up